scripts/deploy.sh

```diff
@@ -26,10 +26,17 @@ ls $DEPLOY_CONF_DIR
 cd $DEPLOY_SERVICE_DIR
 git config --global user.name "osscameroon-bot"
 
-#get repo url takes a `  Fetch URL: git@github.com:elhmn/infra.git`
-#and returns `@github.com/elhmn/infra.git`
-REPO_PATH=$(git remote show origin | grep Fetch | sed -E 's#^.*(git@|https://)([^/]+)/(.*)$#@\2:\3#;s#:#/#g')
-git pull --rebase https://$GITHUB_TOKEN$REPO_PATH main
+if [ ! -z "$GITHUB_TOKEN" ]; then
+	#get repo url takes a `  Fetch URL: git@github.com:elhmn/infra.git`
+	#and returns `@github.com/elhmn/infra.git`
+	REPO_PATH=$(git remote show origin | grep Fetch | sed -E 's#^.*(git@|https://)([^/]+)/(.*)$#@\2:\3#;s#:#/#g')
+	git pull --rebase https://$GITHUB_TOKEN$REPO_PATH main
+else
+	#get repo url takes a `  Fetch URL: git@github.com:elhmn/infra.git`
+	#and returns `@github.com:elhmn/infra.git`
+	REPO_PATH=$(git remote show origin | grep Fetch | sed -E 's#^.*(git@|https://)([^/]+)/(.*)$#@\2/\3#')
+	git pull --rebase git$REPO_PATH
+fi
```

Comment:
Let's simplify this a little bit !

```suggestion
	LINK="https://$GITHUB_TOKEN$(git remote show origin | grep Fetch | sed -E 's#^.*(git@|https://)([^/]+)/(.*)$#@\2:\3#;s#:#/#g') main"
else
	LINK="git$(git remote show origin | grep Fetch | sed -E 's#^.*(git@|https://)([^/]+)/(.*)$#@\2/\3#')"

# for logging purposes
git pull --rebase $LINK
```

What do you think ?

Reply:
of course this is EXTREMLY OPTIONNAL and not needed at all, it's a "fancy estetical" proposal

Reply:
Thank you, I try to not abstract this logic as I am not sure how much they will diverge depending since this script is very immature and could potentially change. Adding an abstraction here that early might cause more arm than anything else 🥲